### PR TITLE
Add high-level e-cash backup&restore documentation

### PR DESCRIPTION
**Pull request by dpc**

### CODE REVIEW

File: docs/backup_and_recovery.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Client E-cash backup&recovery
+
+## Introduction
+
+The client uses [the deterministic e-cash derivation scheme described in](./recoverable_e-cash.md). This allows e-cash recovery from the root key (exported and backed up securely during wallet creation) in case of data corruption or losing access to the wallet.
```
Comment (elsirion) [inline docs/backup_and_recovery.md:5]: described in …?

File: docs/backup_and_recovery.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+The following is the persistent data relevant to the e-cash derivation:
+
+* spendable signed notes
+* unconfirmed notes
+* note derivation index counter for each amount tier
+
+An atomic snapshot of all the above data is taken, along with the current Fedimint consensus `epoch`. The embedded `epoch` is what allows skipping a large part of Federation history during the restoration procedure.
+
+All this data is serialized, padded (to avoid leaking any information), and encrypted with the deterministically derived (from the root seed key) backup encryption key.
```
Comment (elsirion) [inline docs/backup_and_recovery.md:25]: Would be good to add a section for the derivation path here https://github.com/fedimint/fedimint/blob/f3ee5b8e382ed59702227a9a9c7f6be3fcd9ea46/docs/recoverable_e-cash.md#derivation-paths

File: docs/backup_and_recovery.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+The Federation also provides an ability for the client to download the most recent backup snapshot of a given ID.
+
+## Restoring from the snapshot
+
+Backup recovery works by starting from the recent snapshot, then sequentially querying the Federation for consensus epoch history since the backup snapshot was taken until the present time, and for each epoch,  scanning all the relevant consensus data and matching it against the existing (e.g. known spendable notes) and predicted client state (e.g. next in sequence deterministic blind nonces).
+
+In essence, the recovery code is (in limited scope) replaying the Federation consensus history to fast-forward the snapshot to the final and up-to-date state.
+
+The details
```
Comment (elsirion) [inline docs/backup_and_recovery.md:51]: Accidentally left in here?

**Review by elsirion — CHANGES_REQUESTED** [review]
Some nits, I wonder if it would fit in https://github.com/fedimint/fedimint/blob/f3ee5b8e382ed59702227a9a9c7f6be3fcd9ea46/docs/recoverable_e-cash.md rather than being its own doc?

**Comment by elsirion** [comment]
We also need a [section here](https://github.com/fedimint/fedimint/blob/f3ee5b8e382ed59702227a9a9c7f6be3fcd9ea46/docs/recoverable_e-cash.md#key-reduction) for symmetric key derivation.

**Comment by dpc** [comment]
> Some nits, I wonder if it would fit in https://github.com/fedimint/fedimint/blob/f3ee5b8e382ed59702227a9a9c7f6be3fcd9ea46/docs/recoverable_e-cash.md rather than being its own doc

I was wondering the same thing. No strong preferences, though `recoverable_e-cash.md` seems more technical and lower level, while this one seems rather cursory and high level.

**Comment by dpc** [comment]
@elsirion All done.

**Review by elsirion — APPROVED** [review]

**Pull request merged**